src/compressor/lz4/LZ4Compressor.h

```diff
@@ -46,7 +46,7 @@ class LZ4Compressor : public Compressor {
       compressed_len = LZ4_compress_fast_continue(
         &lz4_stream, data, outptr.c_str()+pos, origin_len,
         outptr.length()-pos, 1);
-      if (compressed_len <= 0)
+      if (compressed_len = 0)
```

Comment:
Not sure why the LZ4 function returns int if the result can only be 0 or positive. I assume compressed_len is currently uint32_t because in compressed it's size_t

Reply:
If you prefer " if (compressed_len <= 0)" then we should change the compressed_len var to int

Reply:
@dalgaaf Do you mean "==" ?

Reply:
Sorry, yes! It should be "=="

Reply:
@dalgaaf yes, i prefer `if (compressed_len <= 0)`. so it's more "complete" and avoid implicit castings. but i have no strong feeling on this.